From david-b@pacbell.net  Tue Apr 21 22:40:44 2009
From: David Brownell <david-b@pacbell.net>
Date: Thu, 16 Apr 2009 13:51:46 -0700
Subject: USB: gadget: omap_udc uses platform_driver_probe()
To: Greg KH <greg@kroah.com>
Cc: "Uwe Kleine-König" <u.kleine-koenig@pengutronix.de>
Message-ID: <200904161351.46652.david-b@pacbell.net>


From: David Brownell <dbrownell@users.sourceforge.net>

We now have a more correct solution for shrinking runtime driver
footprints than just marking probe() as __init ... use it.

Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/omap_udc.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/usb/gadget/omap_udc.c
+++ b/drivers/usb/gadget/omap_udc.c
@@ -3104,7 +3104,6 @@ static int omap_udc_resume(struct platfo
 /*-------------------------------------------------------------------------*/
 
 static struct platform_driver udc_driver = {
-	.probe		= omap_udc_probe,
 	.remove		= __exit_p(omap_udc_remove),
 	.suspend	= omap_udc_suspend,
 	.resume		= omap_udc_resume,
@@ -3122,7 +3121,7 @@ static int __init udc_init(void)
 #endif
 		"%s\n", driver_desc,
 		use_dma ?  " (dma)" : "");
-	return platform_driver_register(&udc_driver);
+	return platform_driver_probe(&udc_driver, omap_udc_probe);
 }
 module_init(udc_init);
 
